xc/device.ml

```diff
@@ -608,11 +610,32 @@ let qemu_media_change ~xs device _type params =
 	if is_upstream_qemu device.frontend.domid
 	then begin
 		let cd = "ide1-cd1" in
-		let qmp_cmd =
-			if params = ""
-			then (Qmp.Command(None, Qmp.Eject (cd, Some true)))
-			else (Qmp.Command(None, Qmp.Change (cd, params, None))) in
-		qmp_write device.frontend.domid qmp_cmd
+		match params with
```

Comment:
> We can call qmp_read after sending qmp command.

I think that if we do that, we still have a problem:
- if this doesn't block indefinitely waiting for the next message sent by qemu, it will not detect immediately that a new qmp event is available when qemu sends a message. 
- If this blocks indefinitely waiting for the next message sent by qemu, it will block other qmp_read/write commands.